### ISSUE TITLE
`radio_button` and `checkbox` generate inconsistent markup when `id` is specified

### ISSUE THREAD

**Issue by lcreid** [bootstrap-ruby/bootstrap_form#342]
If the caller specifies a DOM id to `radio_button` or `checkbox`, the generated label's `for` attribute is the "Rails default" id, rather than the custom one. For example, this code:

```ruby
radio_button(:misc, '1', label: 'This is a radio button', id: "custom_id")
```

generates:

```ruby
 <label for="user_misc_1"><input id="custom_id" name="user[misc]" type="radio" value="1"/> This is a radio button</label>
```

You could argue that the expect output should be:

```ruby
<label for="custom_id"><input id="custom_id" name="user[misc]" type="radio" value="1"/> This is a radio button</label>
```

Since the label actually wraps the form control, I don't believe this affects how the page behaves. It does affect the test cases I'm writing for #326, but I can work around the problem in the meantime. I just wanted to put this out in there in case someone sees it as potentially being an issue.

I will submit a pull request with test cases that expose this issue.

**Comment by duleorlovic** [comment]
I write pull requests #357 that will use checkbox `id` to specify label's `for` so I think that your example is covered if pull is merged.

**Comment by lcreid** [comment]
This was fixed by #357. Thanks, @duleorlovic !
